### ISSUE TITLE
Spurious CS8625 warning when checking null against type with equality overloads

### ISSUE THREAD

**Issue by dquist** [dotnet/roslyn#39906]
**Version Used**: 

- Visual Studio 16.3.9
- .NET Core 3.0.100
- C# 8
- Nullable types enabled

**Steps to Reproduce**:

When performing a null check of a value that overloads the equality operators, warning CS8625 is generated. (Cannot convert null literal to non-nullable reference type).

Consider the example below for a class that overloads the equality operators. (The implementations are just stubs but the content doesn't seem to matter).

When attempting to perform a null check in `TestMethod`, the compiler generates warning CS8625. If I comment out the operator overloads the warning goes away.

```c#
public class NullableTestType
{
    public override bool Equals(object? obj) => false;
    public override int GetHashCode() => 0;
    public static bool operator ==(NullableTestType a, NullableTestType b) => a == b;
    public static bool operator !=(NullableTestType a, NullableTestType b) => a != b;

    public static int TestMethod(NullableTestType value)
    {
        if (value == null) // Generates CS8625 when equality operators are included
            throw new ArgumentNullException(nameof(value));

        return value.GetHashCode();
    }
}
```

**Expected Behavior**:

Checking for null equality should be allowed.

**Actual Behavior**:

Warning CS8625 is generated.

A workaround seems to be using the `is` syntax instead of checking for equality, i.e.

```c#
if (value is null) // Doesn't generate an error
    throw new ArgumentNullException(nameof(value));
```

**Comment by RikkiGibson** [comment]
This warning should be avoidable by making the equality operators accept nullable operands.

**Comment by dquist** [comment]
Yes that seems to have resolved it. I was supremely confused but that makes sense.

Thanks!
